File: atcoder.jp/contests/tenka1-2018-beginner/tasks/tenka1_2018_b/main.py
from typing import Tuple

a, b, K = map(int, input().split())


def func(x: int, y: int) -> Tuple[int, int]:
    if x % 2 == 1:
        x -= 1
    tmp = x // 2
    x -= tmp
    y += tmp
    return x, y


for i in range(K):
    if i % 2 == 0:
        a, b = func(a, b)
    else:
        b, a = func(b, a)


print(a, b)

File: atcoder.jp/contests/tenka1-2018-beginner/tasks/tenka1_2018_b/main.rs
use proconio::input;

fn main() {
    input! {
        mut a: usize,
        mut b: usize,
        k: usize,
    }

    for i in 0..k {
        if i % 2 == 0 {
            let (x, y) = func(a, b);
            a = x;
            b = y;
        } else {
            let (y, x) = func(b, a);
            a = x;
            b = y;
        }
    }

    println!("{} {}", a, b);
}

fn func(mut x: usize, mut y: usize) -> (usize, usize) {
    if x % 2 == 1 {
        x -= 1;
    }

    let tmp = x / 2;
    x -= tmp;
    y += tmp;

    (x, y)
}
